### PR TITLE
mesheryctl: Added current-context for system channel command

### LINKED ISSUE #2164: [mesheryctl] View current-context support for `system channel`
#### Current Behavior
Contexts are a new feature in `mesheryctl`. Support for  contexts needs to be added to all `mesheryctl system` commands. See [Meshery CLI Commands](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#) for more "context" (yes, that was a pun).

#### Desired Behavior
`mesheryctl system channel` should return the user’s current subscription: “edge” or “stable”.

#### Implementation
1. Display the “channel” property from the `current-context` in the `~/.meshery/config.yaml` file.

#### Acceptance Tests
1. `mesheryctl system channel` needs to work like `system context`

### PR DESCRIPTION
**Pull request by SudhanAruna**
Signed-off-by: Arunagirinadan Sudharshan <sudhan.aruna.97@gmail.com>

**Description**
Added current-context support for system channel command

This PR fixes #2164 

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
...and here it is!

File: mesheryctl/internal/cli/root/system/channel.go [hunk 1]
```diff
@@ -57,10 +60,15 @@ var channelCmd = &cobra.Command{
...
+
 			log.Info("Subscribed to edge channel.")
 
 		default:
-			log.Fatal("Please subscribe to either the 'stable' or 'edge' release channel.")
+			currentChannel := getCurrentContextChannel()
+			if currentChannel == "" {
+				log.Fatal("Please subscribe to either the 'stable' or 'edge' release channel.")
+			}
+			log.Println("Subscribe dchannel:", currentChannel)
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/channel.go:71]:
```suggestion
			log.Println("Channel subscription:", currentChannel)
```

File: mesheryctl/pkg/utils/helpers.go [hunk 2]
```diff
@@ -86,6 +86,7 @@ var TemplateContext = models.Context{
 	},
 	Platform: "docker",
 	Adapters: ListOfAdapters,
+	Channel:  "",
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/helpers.go:89]:
```suggestion
	Channel:  "stable",
```

`stable` will be the default channel.

**Comment by leecalcote** [comment]
@anirudhjain75 for some reason, I can't add you as a reviewer...

**Comment by anirudhjain75** [comment]
I think it's better to have enum instead of string field for channel. It might be less error prone

**Comment by anirudhjain75** [comment]
Sorry but I'll be able to review this tomorrow . 12hrs from how. Great work though :)

File: mesheryctl/internal/cli/root/system/channel.go [hunk 3]
```diff
@@ -57,10 +60,15 @@ var channelCmd = &cobra.Command{
...
 				return
 			}
+
 			log.Info("Subscribed to edge channel.")
 
 		default:
-			log.Fatal("Please subscribe to either the 'stable' or 'edge' release channel.")
+			currentChannel := getCurrentContextChannel()
+			if currentChannel == "" {
+				log.Fatal("Please subscribe to either the 'stable' or 'edge' release channel.")
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/system/channel.go:69]:
```suggestion
				log.Fatal("No release channel subscription found. Please subscribe to either the 'stable' or 'edge' release channel.")
```

**Comment by leecalcote** [comment]
Scenario: 
- User has config.yaml with current-context defined, but no `channel` attribute.
- User executes `meshery system channel --set stable` and receives this output:

```
 ./mesheryctl system channel --set stable
unable to switch channel with stable
Error subscribing to release channel:unable to set channel
```

The `channel` attribute should be created if one is not found.

**Review by leecalcote — CHANGES_REQUESTED** [review]
@SudhanAruna thanks for this PR. Please have a look and fix this lint issue, if you would - https://github.com/layer5io/meshery/pull/2209/checks?check_run_id=1654184977#step:3:65

**Comment by SudhanAruna** [comment]
> @SudhanAruna thanks for this PR. Please have a look and fix this lint issue, if you would - https://github.com/layer5io/meshery/pull/2209/checks?check_run_id=1654184977#step:3:65

Sorry for that, I'll amend it and push it now.

**Comment by SudhanAruna** [comment]
> Scenario:
> 
> * User has config.yaml with current-context defined, but no `channel` attribute.
> * User executes `meshery system channel --set stable` and receives this output:
> 
> ```
>  ./mesheryctl system channel --set stable
> unable to switch channel with stable
> Error subscribing to release channel:unable to set channel
> ```
> 
> The `channel` attribute should be created if one is not found.

@leecalcote  Working on this. It is raised because when a context is deleted, the meshery.yaml file has last updated channel. When we try to set the same channel again above error is raised.

**Comment by anirudhjain75** [comment]
Hi @SudhanAruna , There is a issue for modulating the docker compose file. I'll be delivering it today. The best thing here is that once that is created, the work here would become easier.

**Comment by SudhanAruna** [comment]
> Hi @SudhanAruna , There is a issue for modulating the docker compose file. I'll be delivering it today. The best thing here is that once that is created, the work here would become easier.
> 
> #2207

@anirudhjain75 Okay

**Review by leecalcote — APPROVED** [review]
Thank you, @SudhanAruna

**Pull request merged**